chia/rpc/wallet_rpc_api.py

```diff
@@ -1421,7 +1426,7 @@ async def nft_get_info(self, request: Dict) -> Optional[Dict]:
         assert self.service.wallet_state_manager is not None
         if "coin_id" not in request:
             return {"success": False, "error": "Coin ID is required."}
-        coin_id = bytes32.from_hexstr(request["coin_id"])
+        coin_id = decode_puzzle_hash(request["coin_id"])
```

Comment:
This one is sort of interesting because the coin ID can either by the launcher ID or some descendent coin ID. At the risk of making things complicated, do you think we could check if the value has the `nft1` prefix, and if so, decode the puzzle hash, otherwise treat as a hex string?

If that's too complicated, I would go with just a hex string instead of bech32m here.